qiskit/transpiler/passes/optimization/_gate_extension.py

```diff
@@ -23,7 +23,7 @@
 try:
     from z3 import Not, And
     HAS_Z3 = True
-except ModuleNotFoundError:
+except ImportError:
```

Comment:
There are a couple more `try: from z3` imports in other files that will need changing too (see #3185)